Linter forcing use of argument in class setter

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:**
3.7.1
* **Node Version:**
6.8.0
* **npm Version:**
3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<!-- Paste your configuration below: -->
```json
{
  "env": {"es6": true, "node": true},
  "extends": "eslint:recommended"
}
```

**What did you do? Please include the actual source code causing the issue.**

<!-- Paste the source code below: -->
```js
exports.Test = class Test {
  set value(v) {
    throw new Error('You must not set the value directly');
  }
};
```

**What did you expect to happen?**
The code should be valid. JavaScript forces a parameter to be declared; the linter should not force us to use it. As far as I can tell object setters behave correctly.

**What actually happened? Please include the actual, raw output from ESLint.**
```
.../test.js
  2:13  error  'v' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/rules/no-unused-vars.js
+++ b/lib/rules/no-unused-vars.js
@@ -494,7 +494,7 @@ module.exports = {
                         if (type === "Parameter") {
 
                             // skip any setter argument
-                            if (def.node.parent.type === "Property" && def.node.parent.kind === "set") {
+                            if ((def.node.parent.type === "Property" || def.node.parent.type === "MethodDefinition") && def.node.parent.kind === "set") {
                                 continue;
                             }
 

--- a/tests/lib/rules/no-unused-vars.js
+++ b/tests/lib/rules/no-unused-vars.js
@@ -256,6 +256,16 @@ ruleTester.run("no-unused-vars", rule, {
             options: [{argsIgnorePattern: "[cd]"}],
             parserOptions: {ecmaVersion: 6},
         },
+
+        // https://github.com/eslint/eslint/issues/7351
+        {
+            code: "(class { set foo(UNUSED) {} })",
+            parserOptions: {ecmaVersion: 6}
+        },
+        {
+            code: "class Foo { set bar(UNUSED) {} } console.log(Foo)",
+            parserOptions: {ecmaVersion: 6}
+        }
     ],
     invalid: [
         { code: "function foox() { return foox(); }", errors: [definedError("foox")] },

